src/main/java/org/sciborgs1155/robot/commands/Climbing.java

```diff
@@ -0,0 +1,50 @@
+package org.sciborgs1155.robot.commands;
+
+import static org.sciborgs1155.robot.Constants.Field.*;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import java.util.function.DoubleSupplier;
+import org.sciborgs1155.robot.drive.Drive;
+
+public class Climbing {
+  private final Drive drive;
+
+  public Climbing(Drive drive) {
+    this.drive = drive;
+  }
+
+  /**
+   * @param coordinates current coordinates of the robot as a Translation2d supplier in meters as
+   *     units.
+   */
+  public Rotation2d findChainAngle(/* Translation2d coordinates */ ) {
+    Translation2d nearest;
+    Translation2d coords = drive.pose().getTranslation();
+
+    nearest = coords.nearest(chainCoordinateList());
+
+    // TODO forgive me lord for I have sinned
+    if (nearest == BLUE_STAGE_AMPSIDE) {
```

Comment:
you can cut this in half by only dealing with the current alliance's stage

Reply:
actually, you can make the chain coordinates `Pose2d`s with the proper rotation inside them, and return the rotation from the nearest pose without the if statement

Reply:
oh my god the entire time it coudlve been a single line i genuinely want to end it all